Analysis/src/EventInfo.cpp

```diff
@@ -52,7 +50,7 @@ const analysis::LepCandidate& EventInfoBase::GetFirstLeg()
 {
     Lock lock(*mutex);
     if(!leg1) {
-        tuple_leg1 = std::make_shared<ntuple::TupleLepton>(*event, GetLegIndex(1));
+        tuple_leg1 = std::make_shared<ntuple::TupleLepton>(event_candidate->GetEvent(), GetLegIndex(1));
```

Comment:
GetFirstLeg shouldn't create any new objects. Use the one created in event_candidate.